tcf_website/models/models.py

```diff
@@ -379,6 +381,31 @@ class Course(models.Model):
     def __str__(self):
         return f"{self.subdepartment.mnemonic} {self.number} | {self.title}"
 
+    def course_description_without_prerequisite(self):
+        """Returns course description without its pre-requisite(s)."""
+        return self.description.replace(self.pre_req, "")
+
+    def prerequisite(self):
+        """Returns course pre-requisite(s) and assigns them to a model attribute pre-req."""
+        if self.description.find("Prerequisite: ") >= 0:
+            pre_req_to_end = self.description[self.description.find("Prerequisite: "):]
+            if pre_req_to_end.find(".") >= 0:  # In line pre-reqs
+                self.pre_req = pre_req_to_end[:pre_req_to_end.find(".") + 1]
+                return self.pre_req[len("Prerequisite: "):-1]
+            # At the end pre-reqs
+            self.pre_req = pre_req_to_end
+            return self.pre_req[len("Prerequisite: "):]
+        if self.description.find("Prerequisites: ") >= 0:
```

Comment:
Given that `"Prerequisites: "` is mostly the same as `"Prerequisite: "`, I'd suggest simply searching for `Prerequisite`, selecting the substring after that, and then dropping everything before/including the first colon. 
That way you don't need this extra if block.

Reply:
Ah very smart move